store/copr/coprocessor.go

```diff
@@ -1061,31 +1142,10 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *tikv.R
 	}
 	var resolveLockDetail *util.ResolveLockDetail
 	if lockErr := resp.pbResp.GetLocked(); lockErr != nil {
-		resolveLockDetail = worker.getLockResolverDetails()
-		// Be care that we didn't redact the SQL statement because the log is DEBUG level.
-		if task.eventCb != nil {
-			task.eventCb(trxevents.WrapCopMeetLock(&trxevents.CopMeetLock{
-				LockInfo: lockErr,
-			}))
-		} else {
-			logutil.Logger(bo.GetCtx()).Debug("coprocessor encounters lock",
-				zap.Stringer("lock", lockErr))
-		}
-		resolveLocksOpts := txnlock.ResolveLocksOptions{
-			CallerStartTS: worker.req.StartTs,
-			Locks:         []*txnlock.Lock{txnlock.NewLock(lockErr)},
-			Detail:        resolveLockDetail,
-		}
-		resolveLocksRes, err1 := worker.kvclient.ResolveLocksWithOpts(bo.TiKVBackoffer(), resolveLocksOpts)
-		err1 = derr.ToTiDBErr(err1)
-		if err1 != nil {
-			return nil, errors.Trace(err1)
-		}
-		msBeforeExpired := resolveLocksRes.TTL
-		if msBeforeExpired > 0 {
-			if err := bo.BackoffWithMaxSleepTxnLockFast(int(msBeforeExpired), errors.New(lockErr.String())); err != nil {
-				return nil, errors.Trace(err)
-			}
+		var err error
+		resolveLockDetail, err = worker.handleLockErr(bo, lockErr, task)
+		if err != nil {
+			return nil, err
 		}
 		return []*copTask{task}, nil
```

Comment:
We should still handle the remaining batch responses and merge them.

Reply:
The same for region error, I think.

Reply:
It is possible that error(lock, region miss, others) is returned in the original response while the batched responses returns `ok`. 
All the lock errors may lead to lock resolving, and all the region errors may lead to region miss retry.
If the order is not quired, we could return the results of the successful responses and not execute them again.

Reply:
The error from `worker.handleLockErr` means that it's failed to resolve lock or a backoff timeout, don't we return the error to the client?

Reply:
No problem in L1148. But if the lock is resolved, I think we shouldn't only return the task itself? `BatchResponses` may include success and failure results and they should be either returned through the channel or returned to retry. (Or is there anything I misunderstand here...?)